### DISCUSSION TITLE
Parsing JSON fields in view collection queries

### DISCUSSION THREAD

**Discussion by aonghas** [pocketbase/pocketbase#4698] [Q&A] [answered]
Hi there,

I have a JSON column in a table called `items` and each record has a JSON array of items, for example:

```
[
  {
    "id": "rrmtksqCJJD4kH93sBVst8",
    "itemTypeId": "b3kf6d0msbgl15a",
    "values": {
      "aHD6f3jQDmr4aMgQr41RcW": null,
      "duration": 300000,
      "siNYYKQWfdQC8U4FoDCrVw": "Details here",
      "title": "Item 1"
    }
  },
  {
    "id": "bfFsWwussJs1XeqkfN3wNQ",
    "itemTypeId": "b3kf6d0msbgl15a",
    "values": {
      "aHD6f3jQDmr4aMgQr41RcW": null,
      "duration": 400000,
      "siNYYKQWfdQC8U4FoDCrVw": "More details",
      "title": "Item 2"
    }
  }
]
```
I was wondering is it possible to build queries for a view collection based on data in JSON values like this?

For example, how would I write a query to sum all the item durations together?

**Comment by ganigeorgiev** [comment, marked as answer]
You can use the SQLite json builtin functions - https://www.sqlite.org/json1.html.

I'm not entirely sure that understand the desired behavior in your case, but here is one example query:

```js
SELECT
    (ROW_NUMBER() OVER()) as id,
    sum(json_extract(`values`, '$.duration')) as totalDuration
FROM items
```

**Reply by aonghas** [reply]
Wow Pocketbase is even cooler than I thought 🤯. Thank you!
